synapse/storage/search.py

```diff
@@ -36,10 +36,12 @@ def search_msgs(self, room_ids, search_term, keys):
         clauses = []
         args = []
 
-        clauses.append(
-            "room_id IN (%s)" % (",".join(["?"] * len(room_ids)),)
-        )
-        args.extend(room_ids)
+        # Make sure we don't explode because the person is in too many rooms.
+        if len(room_ids) > 500:
```

Comment:
If the clauses are omitted do we have a way to ensure that the user only sees the events for the rooms they are in?